### PR TITLE
Syntax highlighting and handling of Geographical area drawing over the ante meridian

### PR DESCRIPTION
**Pull request by pmbohm**
…

### CODE REVIEW

File: src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js [hunk 1]
```diff
@@ -116,16 +116,6 @@ describe('Portal.ui.openlayers.control.SpatialConstraint', function() {
             expect(eventSpy).not.toHaveBeenCalled();
         });
 
-        it('does not clear constraint where viewport area is below minimum', function() {
```
Comment (danfruehauf) [inline src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js]: Why was that test removed?

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 2]
```diff
@@ -14,6 +14,13 @@ Portal.ui.openlayers.SpatialConstraintType = {
 Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.DrawFeature, {
 
     MIN_AREA_PERCENT: 0.01,
+    errorStyle: {
+        fillOpacity:0.3,
```
Comment (danfruehauf) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: Wouldn't you want something like `'0.3px'`? Add a space here anyway after the colon.

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 3]
```diff
@@ -14,6 +14,13 @@ Portal.ui.openlayers.SpatialConstraintType = {
...
+        strokeOpacity:0.5,
```
Comment (danfruehauf) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: Same as above.

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js [hunk 4]
```diff
@@ -142,18 +154,65 @@ Portal.ui.openlayers.control.SpatialConstraint = Ext.extend(OpenLayers.Control.D
         return this.layer.features[0];
     },
 
+    isGeometryLargeEnough: function(geometry) {
```
Comment (danfruehauf) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: Suggest to add tests to this function perhaps?

**Comment by danfruehauf** [comment]
It's looking good. I tested it. It should be enough so we can merge it. See if you would like to add some test coverage on this functionality. I have no idea when we gonna fix it proper, so better have this code tested even though it might be removed sooner than later.

File: web-app/js/portal/ui/openlayers/control/SpatialConstraint.js (hunk 2 above)
Comment (pmbohm) [inline web-app/js/portal/ui/openlayers/control/SpatialConstraint.js]: its opacity, 0.3 = 30% opacity

File: src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js (hunk 1 above)
Comment (pmbohm) [inline src/test/javascript/portal/ui/openlayers/control/SpatialConstraintSpec.js]: the test was removed as it 'does' clear constraint where viewport area is below minimum

**Comment by pmbohm** [comment]
@danfruehauf comments made

**Comment by danfruehauf** [comment]
Cherry picked onto the `openlayers_2_13_1` branch.

**Pull request closed without merging**